File: Easy/2319-check-if-matrix-is-x-matrix/try-3.cpp
/*
 * Problem #2319: Check if Matrix Is X-Matrix
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 19/03/2025, 10:34:25
 * Link: https://leetcode.com/problems/check-if-matrix-is-x-matrix/
 */

class Solution {
public:
    bool checkXMatrix(vector<vector<int>>& grid) {
        int i ,j;
        for(i = 0;i <grid.size();i++){
            for(j = 0;j <grid[0].size();j++){
                if(i == j || i+j==grid.size()-1){
                    if( grid[i][j] == 0){
                        return false;
                    }
                } 
                else{
                    if(grid[i][j] != 0){
                        return false;
                    }
                }
            }
        }
        return true;
    }
};

File: Easy/121-best-time-to-buy-and-sell-stock/try-3.py
/*
 * Problem #121: Best Time to Buy and Sell Stock
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 02/12/2024, 18:40:24
 * Link: https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 */

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        m=s=0
        for i in range(1,len(prices)):
            s=max(s:=s+prices[i]-prices[i-1],0)
            m=max(s,m)
        return m

            

        
